src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -72,13 +80,32 @@ def __init__(self, job_id, initial_params, area, should_export_plots):
     @staticmethod
     def _structure_results_from_area_object(target_area):
         area_dict = dict()
-        area_dict['name'] = target_area.name
+        lst = []
+        if target_area.parent is None:
+            for value in target_area.children:
+                lst.append(value.name)
+            duplicate_values = SimulationEndpointBuffer.get_duplicate_data(lst)
+            SimulationEndpointBuffer.duplicate_list.extend(duplicate_values)
+        if target_area.name in SimulationEndpointBuffer.duplicate_list \
+                and target_area.parent is not None \
+                and target_area.parent.parent is None and target_area.name != 'Market Maker':
+            target_area.rename(target_area.name + target_area.uuid)
+            area_dict['name'] = target_area.name
+        else:
+            area_dict['name'] = target_area.name
```

Comment:
The renaming of the areas should not be part of the simulationEndpointBuffer but should rather be a preprocessing step (even before the simulation starts).

Reply:
Hannes, we can implement this in two places, one is , when we get the request from d3a-web to d3a, we can capture and do it, but for CLI we need to handle in simulationEndpointBuffer only, so instead of doing at two places, i have made the change at common point which utilizes for d3a-web and also for d3a . If we go on with first solution then we need to make changes in different files for passing setupmodule and compare and allow not to apply json_arg , So i felt this is  common point solution which fix both cli and d3a-web

Reply:
The Endpointbuffer is the place were we collect all the results. IMO the renaming of areas should not be part of this, but should be done just before the simulation starts (as stated above) in simulation.py (https://github.com/gridsingularity/d3a/blob/master/src/d3a/d3a_core/simulation.py) There it is also a common place and you don't need to distinguish between d3a-web case and cli case.

Reply:
yes Hannes, area object available in simulation.py also which is the common place also, will move the code.

Reply:
Changes are implemented